lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1269,8 +1270,13 @@ Stmt *StmtChecker::visitBraceStmt(BraceStmt *BS) {
       break;
 
     TC.typeCheckDecl(SubDecl, /*isFirstPass*/false);
+    if (SubDecl->isInvalid())
+      hadError = true;
   }
-  
+
+  if (hadError)
+    return nullptr;
```

Comment:
It looks like when processing returns we create a new AST node, but it's the only one I see in a quick review:

```
return new (TC.Context) FailStmt(RS->getReturnLoc(), nilExpr->getLoc(), RS->isImplicit());
```